metadata/src/main/java/org/apache/kafka/metadata/migration/KRaftMigrationDriver.java

```diff
@@ -391,13 +391,20 @@ void enqueueMetadataChangeEvent(
 
     // Events handled by Migration Driver.
     abstract class MigrationEvent implements EventQueue.Event {
+        // Use no-op handler by default because the handleException will be overridden if needed
+        private Consumer<Throwable> retryHandler = NO_OP_HANDLER;
+
+        public void retryHandler(Consumer<Throwable> retryHandler) {
+            this.retryHandler = retryHandler;
+        }
```

Comment:
Did you consider simply defining an empty `public void retryHandler(Throwable thrown)` that `PollEvent` can override?

Reply:
+1 to this style!

Reply:
Also, should we call `wakeup` (run next poll ASAP)  rather that `scheduleDeferred` if the exception is retryable?